app/src/main/java/com/android/unio/model/user/User.kt

```diff
@@ -58,3 +58,55 @@ data class User(
 ) {
   companion object
 }
+
+enum class UserSocialError(val errorMessage: String) {
+  EMPTY_FIELD("The input is empty or blank"),
+  INVALID_PHONE_NUMBER("The phone number has wrong format"),
+  INVALID_WEBSITE("The website is not encoded with https"),
+  NONE("")
+}
+// Helper methods
+/**
+ * @return NONE: no problem is found
+ * @return EMPTY_FIELD: the website is not encoded with https
+ * @return INVALID_PHONE_NUMBER: the input is empty or blank
+ * @return INVALID_WEBSITE: the phone number has wrong format
+ */
+fun checkSocialContent(userSocial: UserSocial): UserSocialError {
+
+  if (userSocial.content.isEmpty() || userSocial.content.isBlank()) {
+    return UserSocialError.EMPTY_FIELD
+  }
+
+  when (userSocial.social) {
+    Social.WHATSAPP -> {
+      val numberRegex = Regex("^[0-9]{10,12}")
```

Comment:
The regex for phone numbers might be too strict or not account for all valid formats across different countries. Consider allowing optional country codes or using a more flexible regex pattern.

Reply:
update : as @oskar-codes told me, it's actually a requirement from WhatsApp. Sorry, the placeholder you made is actually very clear.